Yoast/YoastSEO.js
# Add stemmer and basic noun morphology functionality

## Summary

This PR can be summarized in the following changelog entry:

* [not-user-facing] Adds a German stemmer and functionality to create German noun forms.

## Relevant technical choices:

* The stemmer was adapted from an existing stemmer; authorship was attributed accordingly. The code was adapted from this specific [JavaScript implementation](http://snowball.tartarus.org/otherlangs/german_javascript.txt) of an [existing German stemming algorithm](http://snowball.tartarus.org/algorithms/german/stemmer.html).

## Test instructions

This PR can be tested by following these steps:

* Make sure you have the premium configuration set to the right branch (see https://github.com/Yoast/YoastSEO.js-premium-configuration/pull/7).
* Change the language of your site to German.
* Add a German text and a keyphrase (consisting of German nouns and function words; verbs and adjectives aren't supported yet).
* Make sure that all forms of the keyphrase that occur in the text are found. More extensive testing of various word forms is done by @christinatsaousi 

Fixes #2049
